File: frontend/src/types/persona.ts
export type PersonaRole = 'moderator' | 'petitioner' | 'resident' | 'council_member';

export type PersonaArchetype =
  | 'concerned_parent'
  | 'environmental_activist'
  | 'property_owner'
  | 'local_business_owner'
  | 'fiscal_conservative'
  | 'longtime_resident';

export interface Persona {
  id: string;
  name: string;
  role: PersonaRole;
  archetype?: PersonaArchetype;
  age?: number;
  occupation: string;
  background: string;
  speaking_style: string;
  primary_concern: string;
  secondary_concerns: string[];
  intensity: number;
  color: string;
}

File: backend/models/persona.py
from pydantic import BaseModel, Field
from enum import Enum
from typing import Optional


class PersonaRole(str, Enum):
    MODERATOR = "moderator"
    PETITIONER = "petitioner"
    RESIDENT = "resident"
    COUNCIL_MEMBER = "council_member"


class PersonaArchetype(str, Enum):
    CONCERNED_PARENT = "concerned_parent"
    ENVIRONMENTAL_ACTIVIST = "environmental_activist"
    PROPERTY_OWNER = "property_owner"
    LOCAL_BUSINESS_OWNER = "local_business_owner"
    FISCAL_CONSERVATIVE = "fiscal_conservative"
    LONGTIME_RESIDENT = "longtime_resident"


class Persona(BaseModel):
    id: str
    name: str
    role: PersonaRole
    archetype: Optional[PersonaArchetype] = None
    age: Optional[int] = None
    occupation: str = ""
    background: str = ""
    speaking_style: str = ""
    primary_concern: str = ""
    secondary_concerns: list[str] = Field(default_factory=list)
    intensity: int = Field(default=5, ge=1, le=10, description="How strongly they feel (1-10)")
    system_prompt: str = ""
    color: str = Field(default="#6366f1", description="Display color for UI")
